### DISCUSSION TITLE
When are you going to support 3rd party libraries?

### DISCUSSION THREAD

**Discussion by racktime** [vuepress-theme-hope/vuepress-theme-hope#1156] [Q&A] [answered]
When will you support 3rd party libraries?

**Comment by Mister-Hope** [comment]
What do you mean by 3rd party libraries

**Reply by racktime** [reply]
Vue 3rd party component so that it can be used in the demo block

**Reply by Mister-Hope** [reply]
> Vue 3rd party component so that it can be used in the demo block

I relook your original discussion.

I peosonally recommend you to use your component directly. And copy your script to a code block to show to actual code.

**My code demo feature is targeting runtime, and it’s fully decoupled with the Vue instance in VuePress.** So that if you want to use a component in Vue, you should try to install it on Vue instance in runtime( that is `window.Vue` Imported by CDN links in head)

**Comment by racktime** [comment]
Markdown Enhance json config  is not support ?

```json
// json block is for config
{
  // your config here (optional)
   "jsLib": [
    "https://cdn.jsdelivr.net/npm/echarts/dist/echarts.min.js",
    "https://cdn.jsdelivr.net/npm/v-charts/lib/line.min.js"
  ]
}
```

**Reply by Mister-Hope** [reply]
JSON do not support comment.

**Comment by racktime** [comment]
::: demo [vue] A Vue Demo

```vue
<template>
  <div class="box">
i want using custom component
  </div>
</template>
<script>
export default {
  data: () => ({ message: "very handsome" }),
  methods: {
    handler() {
      alert(this.message);
    },
  },
};
</script>
<style>
.box span {
  color: red;
}
</style>
```

:::

**Reply by Mister-Hope** [reply]
I never said "vue" block is supported. The document is detailed enough, if you cannot catch the docs idea, maybe you can search something else as a replacement.

**Comment by Mister-Hope** [comment, marked as answer]
Can not be supported on plugin side due to VuePress structure design.
